test/testdata/infer/autocorrect_t_let_nilable.rb

```diff
@@ -0,0 +1,5 @@
+# typed: true
+
+x = T.let(5, T.nilable(Integer))
+  T.let(x, Integer)
+# ^^^^^^^^^^^^^^^^^ error: Argument does not have asserted type `Integer`
```

Comment:
not blocking but I typically write these tests like this:
```suggestion
extend T::Sig

sig {params(x: T.nilable(Integer)).void}
def example(x)
  T.let(x, Integer)
# ^^^^^^^^^^^^^^^^^ error: Argument does not have asserted type `Integer`
end
```

- Sometimes you can get into weird behavior because `T.let` is both a way to annotate a type and also to pin a variable's type in a loop. This test isn't one of them, but i still tend to write it differently
- The indentation of code inside the method naturally makes room to write the `^^^` assertion